src/app/interfaces/storage.interface.ts

```diff
@@ -114,3 +114,71 @@ export interface DiskUpdate {
   informational?: number;
   enclosure?: EnclosureSlot;
 }
+
+/**
+ * As returned by snapshot.query
+ */
+export interface SnapshotData {
+  name: string;
+  snapshot: string;
+  dataset: string;
+  created?: string;
+  properties?: SnapshotProperties;
+  referenced?: string;
+  used?: string;
+}
+
+export interface SnapshotProperty {
+  parsed: string;
+  rawvalue: string;
+  value: string;
+  source: string;
+}
```

Comment:
I've added something called ZfsProperty, usage is ZfsProperty<number> or ZfsProperty<string> based on parsed value.

Reply:
Switched to ZfsProperty interface